### PR TITLE
[tests] make sure Kubevirt resource remains consistent during e2e run

### PR DESCRIPTION
**Pull request by enp0s3**
Signed-off-by: Igor Bezukh <ibezukh@redhat.com>

**What this PR does / why we need it**:
In order to prevent leftovers or any other unintended changes
in Kubevirt resource this patch suggests to check the resource
in BeforeTestCleanup.

The original Kubevirt resource that comes with the kubevirt cluster
will be stored in OriginalCR var. When the whole e2e test suite ends
the original kv resource will be restored. During the test suites run
the BeforeTestCleanup will check if the kv resource was changed. If it
was changed it will be restored either to the original or the the so called
e2e default configuration (that is always used by the automation)

**Special notes for your reviewer**:
I know we can refactor that area. It will come in the following PRs. The aim of this PR is to
give initial functional solution.

```release-note
NONE
```

### CODE REVIEW

**Review by jean-edouard — APPROVED** [review]

File: tests/utils.go [hunk 1]
```diff
@@ -4533,6 +4548,7 @@ func resetToDefaultConfig() {
 		return
 	}
 
+	RestoreToE2eKubevirtResource()
 	UpdateKubeVirtConfigValueAndWait(KubeVirtDefaultConfig)
```
Comment (rmohr) [inline tests/utils.go:4552]: This looks to me like we should have two ways of setting something on the KubeVirt CR:

 1. with `UpdateKubeVirtConfigValueAndWait(config)` which people would use in their tests
 2. the general cleanup in this function after each tests. `UpdateKubeVirtCRAndWait(e2ecr)`
 
 It may not be an issue or not, but just looking at this code I am not sure if we would wait int all cases that the changes got received by all components. It may for instance be the case that `RestoreToE2eKubevirtResource()` would revert the CR and `UpdateKubeVirtConfigValueAndWait` would think that the CR is already in sync and it would not check all components.

To sum up, it may make sense to collaps the two calls here to one function which reverts the whole resource and waits for  it to be propagated.

One additional note: I would have expected that only the operator tests are changing the KV resource outside of the configuration section. Let's just ensure that the revert stays efficient.

Comment (enp0s3) [reply tests/utils.go:4552]: @rmohr Currently there is a test in `vmi_lifecycle_test` that abuses the infra placement so the virt-handler will be evicted during VM run. If you have some alternative way to evict or permanently kill virt-handler for a period of time, it will be great because then indeed only the operator tests will touch the KV CR

Comment (enp0s3) [reply tests/utils.go]: @rmohr PTAL at the new changes. Currently I don't have another way to suggest in VMI lifecycle test, so that we won't that the KV CR there.
